contracts/ERC677BridgeToken.sol

```diff
@@ -8,17 +8,29 @@ import "./interfaces/IBurnableMintableERC677Token.sol";
 import "./upgradeable_contracts/Claimable.sol";
 
 contract ERC677BridgeToken is IBurnableMintableERC677Token, DetailedERC20, BurnableToken, MintableToken, Claimable {
-    address public bridgeContract;
+    address[] internal _bridgeContracts;
```

Comment:
Although I don't expect to have a lot of bridges, I agree with you that it should use the linked-list approach. But since there is `MAX_VALIDATORS` limit of `50` items and `_removeValidator` requires looping through the list, I'm not sure this approach would significantly win in gas. However, it could win in storage space (because the list approach uses one uint256 field and one mapping against my approach which uses an array and a mapping), but in case of a small number of bridges it doesn't matter.

Reply:
Also, I assume that `setBridgeContracts` is a rare operation.

Reply:
In fact, any array in Solidity requires N+1 storage slots: N storage  for the N array elements and 1 element to store the array index/length. So, storing the one bridge address in the linked list has the same efficiency as your approach, storing two bridge addresses in the linked list will consume less gas (by 1 storage slot), storing 3 bridge addresses -- by 2 storage slots etc. Here we are not considering the case that a transaction to send an array with one element requires more bytes since the array is RLP-encoded, so, even to decode the input data EVM will spend more resources.

Reply:
BTW, `50` is a limitation which was asked to be added on one of the previous security audit.  Bearing in the mind the same idea, it makes sense to limit the size of the bridge addresses list.

Reply:
Ok, I will change the code making it similar to the linked-list approach you are proposing.